react/javascript/src/components/app/RuntimeIcon.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { VoidFunctionComponent } from 'react'
+import { NodeJs } from './icons/NodeJs'
+import { Jvm } from './icons/Jvm'
+import { Ruby } from './icons/Ruby'
+
+export const RuntimeIcon: VoidFunctionComponent<{ name: string }> = ({ name }) => {
+  // TODO need to pattern match more permissively here
```

Comment:
As other comment, needs to be better, do we have prior art?

Reply:
```tsx
  if (!name) {
    return <span>Unknown Runtime</span>
  }

  if (name.match(/(oracle|openjdk|java)/i)) {
    return <Java />
  }
  if (name.match(/ruby/i)) {
    return <Ruby />
  }
  if (name.match(/node/i)) {
    return <JavaScript />
  }

  return <span>{name}</span>
```